### PR TITLE
Fix Go 1.11 incompatibilities

### LINKED ISSUE #1039: Compilation error with Go 1.11
## Requirement - what kind of business use case are you trying to solve?

A compilation error occurs when running `make` on the main project using Go 1.11.

## Problem - what in Jaeger blocks you from solving the requirement?

The forwarding of vararg arguments to `printf` wrapper functions without explicit conversion to a slice. As the release notes of Go 1.11 point out:

> Additionally, go vet has become more robust when format-checking printf wrappers.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

~~Use explicit casting to convert varargs to a slice for use in `printf` wrappers.~~
Avoid use of varargs as a slice. Expand varargs when forwarding to another function.

### PR DESCRIPTION
**Pull request by isaachier**
Signed-off-by: Isaac Hier <isaachier@gmail.com>

## Which problem is this PR solving?
- Fixes #1039.

## Short description of the changes
- Use casting to suppress error.

### CODE REVIEW

File: plugin/storage/cassandra/samplingstore/storage_test.go [hunk 1]
```diff
@@ -420,8 +420,22 @@ func TestStringToProbabilitiesAndQPS(t *testing.T) {
 	probabilities := s.stringToProbabilitiesAndQPS(testStr)
 
 	assert.Len(t, probabilities, 2)
-	assert.Equal(t, map[string]*model.ProbabilityAndQPS{"GET": {0.001, 63.2}, "PUT": {0.002, 0.0}}, probabilities["svc1"])
-	assert.Equal(t, map[string]*model.ProbabilityAndQPS{"GET": {0.5, 34.2}}, probabilities["svc2"])
+	assert.Equal(t, map[string]*model.ProbabilityAndQPS{
```
Comment (isaachier) [inline plugin/storage/cassandra/samplingstore/storage_test.go]: One of the tools for Go 1.11 doesn't like structs without field names.

File: pkg/recoveryhandler/zap.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ type zapRecoveryWrapper struct {
 // Println logs an error message with the given fields
 func (z zapRecoveryWrapper) Println(fields ...interface{}) {
 	// if you think i'm going to check the type of each of the fields and then logger with fields, you're crazy.
-	z.logger.Error(fmt.Sprintln(fields))
+	z.logger.Error(fmt.Sprintln([]interface{}(fields)))
```
Comment (yurishkuro) [inline pkg/recoveryhandler/zap.go]: don't we want `fields...` instead here?

Comment (isaachier) [reply pkg/recoveryhandler/zap.go]: Well, the existing test case won't work out that way. It expects `Println(1, 2, 3)` to log as `message: [1, 2, 3]`, not `message: 1, 2, 3`. I'm just wondering why one of the `zap` convenience wrappers weren't used here. Might replace.

Comment (yurishkuro) [reply pkg/recoveryhandler/zap.go]: looking at gorilla's interface:
https://github.com/gorilla/handlers/blob/7e0847f9db758cdebd26c149d0ae9d5d0b9c98ce/recovery.go#L10

I don't think it implies that if you call `logger(1, 2, 3)` that you would get `[1, 2, 3]` output. Rather, it should be equivalent to calling `println(1, 2, 3)`, which will print `1 2 3`.

Comment (yurishkuro) [reply pkg/recoveryhandler/zap.go]: basically, our implementation of the adapter was incorrect

**Pull request merged**